ToDoApp/ToDoApp/ToDoListView.swift

```diff
@@ -51,26 +49,29 @@ struct ToDoListView: View {
                             .frame(minWidth: Constants.addButtonSize, minHeight: Constants.addButtonSize)
                     }
                     .listStyle(.plain)
-                }
-                Button( action: {
-                    viewModel.addButtonTapped()
-                }, label: {
-                    Image(systemName: "plus")
-                        .frame(minWidth: Constants.addButtonSize, minHeight: Constants.addButtonSize)
-                        .foregroundColor(.white)
-                        .background(Color(.primaryButton))
-                        .clipShape(Circle())
-                }
-                )
-                .padding(.trailing, 17)
-                .sheet(item: Binding(
-                    get: { viewModel.destination },
-                    set: { _ in viewModel.resetDestination()}
-                )
-                ) { destination in
-                    switch destination {
-                    case .addTask(let viewModel):
-                        CreateTaskView(viewModel: viewModel)
+                    if viewModel.loading {
+                        ProgressView()
+                    }
```

Comment:
this should be after the switch statement, not inside. if it is inside, then there would be no progress view during initial loading of the view when it fetches the todo models the first time

Reply:
This has been fixed.